pywayland/scanner/protocol.py

```diff
@@ -68,45 +68,83 @@ def parse(cls, element: ET.Element) -> Protocol:
     def __repr__(self) -> str:
         return f"Protocol({self.name})"
 
-    def output(self, output_dir: str, module_imports: dict[str, str]) -> None:
+    def output(self, output_dir: str, all_imports: dict[str, str]) -> None:
         """Output the scanned files to the given directory
 
         :param output_dir: Path of directory to output protocol files to
         :type output_dir: string
         """
-        protocol_name = self.name.replace("-", "_")
-
-        output_dir = os.path.join(output_dir, protocol_name)
         if not os.path.exists(output_dir):
             os.mkdir(output_dir)
 
-        # First, we'll create the __init__.py file
-        printer = Printer(protocol_name)
+        protocol_name = self.name.replace("-", "_")
+        protocol_path = os.path.join(output_dir, f"{protocol_name}.py")
+
+        printer = Printer(protocol_name, all_imports)
         if self.copyright:
             self.copyright.output(printer)
         else:
             printer(copyright_default)
+        printer()
 
+        printer("from __future__ import annotations")
         printer()
-        for iface in sorted(self.interface, key=lambda x: x.name):
-            printer(f"from .{iface.name} import {iface.class_name}  # noqa: F401")
 
-        init_path = os.path.join(output_dir, "__init__.py")
-        with open(init_path, "wb") as f:
-            printer.write(f)
+        if any([iface.enum for iface in self.interface]):
+            printer("import enum")
+            printer()
+
+        typing_imports = []
+        if any([iface.needs_any_type for iface in self.interface]):
+            typing_imports.extend(["Any"])
+
+        pywayland_imports = []
+        if any([iface.needs_t_type for iface in self.interface]):
+            typing_imports.extend(["TypeVar"])
 
-        # Now build all the modules
+        if typing_imports:
+            printer(f"from typing import {', '.join(sorted(typing_imports))}")
+            printer()
+
+        if any([iface.needs_argument_type for iface in self.interface]):
```

Comment:
```suggestion

        pywayland_imports = ["Interface", "Global", "Proxy", "Resource"]
        if any(iface.needs_argument_type for iface in self.interface):
```
Move the `pywayland_imports` declaration down here, and set it up with the items that will always be in it.